### ISSUE TITLE
Cannot add new Cluster Profile

### ISSUE THREAD

**Issue by delhora** [gocd/kubernetes-elastic-agents#159]
Hello
I've been testing goCD, and I'm getting stuck with some issues.

**Scenario**
* One dev K8s cluster with goCD installed via helm
* One staging k8s cluster that I'm trying to connect as Elastic Cluster profile.

**Steps to Reproduce**

1) Install goCD via helm in the dev cluster

`helm install stable/gocd --name gocd --namespace gocd`

2) Go to Staging Cluster and create service account

`kubectl create sa gocd`

3) Create Clusterrole 
```
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
  labels:
    app: gocd
    chart: gocd-1.19.3
    heritage: Tiller
    release: gocd
  name: gocd
  resourceVersion: "13566932"
  selfLink: /apis/rbac.authorization.k8s.io/v1/clusterroles/gocd
  uid: ec92d205-0d13-11ea-b5b7-42010a840176
rules:
- apiGroups:
  - ""
  resources:
  - pods
  - pods/log
  verbs:
  - '*'
- apiGroups:
  - ""
  resources:
  - nodes
  verbs:
  - get
  - list
- apiGroups:
  - ""
  resources:
  - events
  verbs:
  - list
  - watch
- apiGroups:
  - ""
  resources:
  - namespaces
  verbs:
  - get
```
 4) Create a Clusterrolebinding for new ClusterRole

`k create clusterrolebinding gocd --clusterrole gocd --serviceaccount <namespace>:gocd`

5) Go to **Admin > Elastic Profiles > Add Cluster Profile**
6) Add a new id (ie. Test)
7) Add the Go Server URL
8) Add cluster IP getting it with `kubectl cluster-info`
9) Namespace: <namespace> configured in the clusterrolebinding
10) Security Token: The one from the secret in `k describe sa gocd`
11) Cluster CA: Same as above

Once I save it and try to Click "Status Report" I get the following message:

` Failure executing: GET at: https://XX.XX.XX.XX/api/v1/nodes. Message: Unauthorized! Configured service account doesn't have access. Service account may have been revoked. Unauthorized.`

If I try to 

`kubectl  auth can-i list nodes --namespace <namespace> --as=system:serviceaccount:<namespace>:gocd ` I get as result `yes`

I'm a bit lost, and I would gladly accept some helm on that :) 

Thanks in advance!

PS: Every time I try to update the cluster information I got this Error: 
`Someone has modified the configuration for Cluster Profile 'stg'. Please update your copy of the config with the changes and try again.`

Don't know if it's related to configuration or just another random issue.

**Comment by arvindsv** [comment]
@delhora In the staging cluster, you haven't really created the `gocd` namespace. The namespace that the service account is created is `default`.

You should be able to see that by doing this:

```
$ kc describe sa gocd
Name:                gocd
Namespace:           default
Labels:              <none>
Annotations:         <none>
Image pull secrets:  <none>
Mountable secrets:   gocd-token-vrm8t
Tokens:              gocd-token-vrm8t
Events:              <none>
```

You can also check the permissions using this:

```
$ kubectl auth can-i list nodes --namespace gocd --as=system:serviceaccount:gocd:gocd
Warning: resource 'nodes' is not namespace scoped
no

$ kubectl auth can-i list nodes --namespace gocd --as=system:serviceaccount:default:gocd
Warning: resource 'nodes' is not namespace scoped
no
```

So, your `clusterrolebinding` should also give the `gocd` cluster role permissions to the service account `default:gocd` and not `<namespace>:gocd`. Something like this:

```
kc create clusterrolebinding gocd --clusterrole gocd --serviceaccount default:gocd
```

When I did that, the permissions changes:


```
$ kubectl auth can-i list nodes --namespace gocd --as=system:serviceaccount:default:gocd
Warning: resource 'nodes' is not namespace scoped
yes
```

The status report started working after that. Can you see if that makes sense and works for you too? Thanks.

If not, can you check in the plugin log file, and see if there are any other errors?

**Comment by delhora** [comment]
Hello!

Thanks for your reply :D 

I created the `sa` in default namespace

```
$ kubectl describe sa gocd
Name:                gocd
Namespace:           default
Labels:              <none>
Annotations:         <none>
Image pull secrets:  <none>
Mountable secrets:   gocd-token-5dx6c
Tokens:              gocd-token-5dx6c
Events:              <none>
```

And I've created the rolebinding 

```
$ kubectl create clusterrolebinding gocd --clusterrole gocd --serviceaccount default:gocd
```

```
$  kubectl describe clusterrolebindings.rbac.authorization.k8s.io gocd
Name:         gocd
Labels:       <none>
Annotations:  <none>
Role:
  Kind:  ClusterRole
  Name:  gocd
Subjects:
  Kind            Name  Namespace
  ----            ----  ---------
  ServiceAccount  gocd  default
```

If I do the test I can get a `yes`

```
$  kubectl auth can-i list nodes --namespace gocd --as=system:serviceaccount:default:gocd
yes
```

After configured the proper token and ca I got the same error in Status Report

` Message: Unauthorized! Configured service account doesn't have access. Service account may have been revoked. Unauthorized`

Thanks again for helping me out!

**Comment by arvindsv** [comment]
@delhora Strange. I assume you've used the right token from `kubectl describe secret gocd-token-5dx6c` and the right cluster info to the staging cluster. When I try this out, it works fine.

I don't think it matters, but the cluster URL I used was something like: `https://aa.aa.aa.aa` and I used the namespace `gocd`. I tried it on GoCD 19.10.0 and plugin version: 3.4.0-196.

Maybe someone else has an idea.
